src/bio/bio_device.c

```diff
@@ -699,7 +699,12 @@ led_device_action(void *ctx, struct spdk_pci_device *pci_device)
 		return;
 
 	pci_dev_type = spdk_pci_device_get_type(pci_device);
-	if (strcmp(pci_dev_type, "vmd") != 0) {
+	if (pci_dev_type == NULL) {
+		D_ERROR("nil pci device type returned\n");
+		opts->status = -DER_MISC;
```

Comment:
Might go with -DER_INVAL for this, since getting NULL back seems like it implies there's something wrong with the pci_device input.

Reply:
this is a "should never happen" situation, the SPDK API should never return NULL here so DER_INVAL doesn't seem like the right fit as to me it implies some form of input is invalid

Reply:
If it should never happen, maybe it should be a D_ASSERT